openoa/utils/downloader.py

```diff
@@ -0,0 +1,425 @@
+        
+        file_name = f["key"]
+                
+        outfile = outfile_path+file_name
+        
+        
+        # check if file exists
+        if os.path.exists(outfile):
+            
+            
+            # if it does check the checksum is correct
+            with open(outfile, "rb") as f_check:
+                file_hash = hashlib.md5()
+                while chunk := f_check.read(8192):
+                    file_hash.update(chunk)
+        
+            if f["checksum"][4:]==file_hash.hexdigest():
+                logger.info("File already exists: " + file_name)
+            
+            
+            # download and unzip if the checksum isn't correct
+            else:
+                
+                logger.info("Downloading: " + file_name)
+                logger.info("File size: " + str(round(f["size"]/(1024*1024),2)) + "MB")       
+
+                download_file(url_file,outfile)
+
+                logger.info("Saved to: " + str(outfile) + "\n")
+
+                if outfile.endswith(".zip"):
+                    with ZipFile(outfile) as zipfile:
+                        zipfile.extractall(outfile_path)
+        
+        
+        # download and unzip if the file doesn't exist
+        else:
+            
+            logger.info("\nDownloading: " + file_name)
+            logger.info("File size: " + str(round(f["size"]/(1024*1024),2)) + "MB")       
+
+            download_file(url_file,outfile)
+
+            logger.info("Saved to: " + str(outfile) + "\n")
+
+            if outfile.endswith(".zip"):
+                with ZipFile(outfile) as zipfile:
+                    zipfile.extractall(outfile_path)
+
+
+
+def get_era5(
+        lat,
+        lon,
+        save_pathname,
+        save_filename,
+        ):
+    """
+    Get ERA5 data directly from the CDS service
+    This requires registration on the CDS service
+    See: https://cds.climate.copernicus.eu/api-how-to
+
+    Monthly 10m height data is demonstrated here, as hourly data takes too long to download, 
+    but could be amended and other CDS datasets also used (e.g. CERRA for Europe)
+
+    Args:
+        lat(:obj:`float`): Latitude in WGS 84 spatial reference system (decimal degrees)
+        lon(:obj:`float`): Longitude in WGS 84 spatial reference system (decimal degrees)
+        save_pathname(:obj:`string`): The path where the downloaded reanalysis data will be saved
+        save_filename(:obj:`string`): The file name used to save the downloaded reanalysis data
+
+    Returns:
+        df(:obj:`dataframe`): A dataframe containing time series of the requested reanalysis variables
+        Saved NetCDF annual ERA5 files
+        Saved ERA5 csv file
+    """
+
+    logger.info("Please note access to ERA5 data requires registration")
+    logger.info("Please see: https://cds.climate.copernicus.eu/api-how-to")
+
+    # set up cds-api client
+    try:
+        c = cdsapi.Client()
+    except Exception as e:
```

Comment:
This would fail naturally, but I wanted to add some additional guidance for people so they know where to look to get help. For example the typical error, "Exception: Missing/incomplete configuration file: C:\Users\Charlie/.cdsapirc," is unlikely to be much help to people that have reached that point. So I've added direction to the help page.

Raises added to docstring